Ginger/GingerCore/ALM/QtestCore.cs

```diff
@@ -0,0 +1,657 @@
+                                                                                    new List<QTestApiModel.PropertyResource>());                   
+                    testSuite = testsuiteApi.CreateTestSuite((long)Convert.ToInt32(ALMCore.DefaultAlmConfig.ALMProjectKey), testSuite, (long)Convert.ToInt32(parentObjectId), "test-cycle");
+
+                    foreach (ActivitiesGroup ag in businessFlow.ActivitiesGroups)
+                    {
+                        QTestApiModel.TestRunWithCustomFieldResource testRun = new QTestApiModel.TestRunWithCustomFieldResource(null, null, null, null,
+                                                                                                                                ag.Name,
+                                                                                                                                new List<QTestApiModel.PropertyResource>(),
+                                                                                                                                testcaseApi.GetTestCase((long)Convert.ToInt32(ALMCore.DefaultAlmConfig.ALMProjectKey), (long)Convert.ToInt32(ag.ExternalID))); 
+                        testRun = testrunApi.Create((long)Convert.ToInt32(ALMCore.DefaultAlmConfig.ALMProjectKey), testRun, testSuite.Id, "test-suite");                      
+                    }     
+                }
+                else
+                {
+                    //##update existing TestSuite
+                    QTestApiModel.TestSuiteWithCustomFieldResource testSuiteToUpdate = testsuiteApi.GetTestSuite((long)Convert.ToInt32(ALMCore.DefaultAlmConfig.ALMProjectKey), (long)Convert.ToInt32(mappedTestSuite.ID));
+
+                    //foreach (QtestTest test in mappedTestSuite.Tests)
+                    //{
+                        //    ActivitiesGroup ag = businessFlow.ActivitiesGroups.Where(x => (x.ExternalID == tsTest.TestId.ToString() && x.ExternalID2 == tsTest.ID.ToString())).FirstOrDefault();
+                        //    if (ag == null)
+                        //        testsF.RemoveItem(tsTest.ID);
+                        //    else
+                        //        existingActivitiesGroups.Add(ag);
+                    //}
+                }
+
+                //set item fields
+                //foreach (ExternalItemFieldBase field in testSetFields)
+                //{
+                //    if (field.ToUpdate || field.Mandatory)
+                //    {
+                //        if (string.IsNullOrEmpty(field.SelectedValue) == false && field.SelectedValue != "NA")
+                //            testSet[field.ID] = field.SelectedValue;
+                //        else
+                //            try { testSet[field.ID] = "NA"; }
+                //            catch { }
+                //    }
+                //}                 
+
+                businessFlow.ExternalID = testSuite.Id.ToString();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['testSuite' is null on at least one execution path.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5395796)